makefiles/libc/newlib.mk

```diff
@@ -1,7 +1,15 @@
+ifeq ($(BOARD), native)
+  $(error Newlib is not supported on native target)
+endif
+
 ifneq (,$(filter newlib_nano,$(USEMODULE)))
+  USE_NEWLIB_NANO = 1
   # Test if nano.specs is available
   ifeq ($(shell $(LINK) -specs=nano.specs -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
-    USE_NEWLIB_NANO = 1
+    # Clang/llvm has no support for specs files
```

Comment:
Shouldn't the test above already filter out any compiler which errors on -specs?

Reply:
Even though it has no support for it, Clang does not fail when passed the  -specs option. Adding -Werror makes sure Clang fails when the -specs flag is passed.